From eb8a3045a5e2aef1c804b49aba3098bac258dbd1 Mon Sep 17 00:00:00 2001
From: Sagar Cheluvegowda <quic_scheluve@quicinc.com>
Date: Fri, 13 Dec 2024 20:38:00 -0800
Subject: [PATCH 09/28] PENDING: net: stmmac: Move desc cache compute within
 HDMA supported flag

Move desc cache compute within HDMA supported flag to ensure
it gets called only in case when HDMA is enabled.

Change-Id: I8013dcefca30342f0b2570daf09062cbebbcd7ef
Signed-off-by: Sagar Cheluvegowda <quic_scheluve@quicinc.com>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
index f112251be2933..fd2e85ee7876a 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -3323,9 +3323,9 @@ static int stmmac_init_dma_engine(struct stmmac_priv *priv)
 		/* DMA TX Mapping Configuration for offline channels */
 		for (chan = tx_channels_enabled; chan < tx_channels_supported; chan++)
 			stmmac_map_tx_offline_chan(priv, priv->ioaddr, priv->plat->dma_cfg, chan);
-	}
 
-	stmmac_desc_cache_compute(priv, priv->ioaddr);
+		stmmac_desc_cache_compute(priv, priv->ioaddr);
+	}
 
 	return ret;
 }
